src/format/mod.rs

```diff
@@ -357,6 +357,7 @@ impl ParseError {
 
 /// The category of parse error
 #[derive(Debug, Clone, PartialEq, Eq, Copy, Hash)]
+#[non_exhaustive]
```

Comment:
Can you have a look at the other uses of `non_exhaustive` on main, and backport them to 0.4.x?

Reply:
There is only one other case, on `Parsed`. But I have my doubts on that one, because of [this comment](https://github.com/chronotope/chrono/blob/0.4.x/src/format/parsed.rs#L108):
```rust
    /// A dummy field to make this type not fully destructible (required for API stability).
    _dummy: (),
```

Reply:
Sorry, reading it again the comment is just another way of saying the same thing...

Reply:
Are you worried that `non_exhaustive` doesn't prevent destruction? I think it does...